compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2514,7 +2514,7 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
         def fullName: String = self.denot.fullName.toString
 
         def pos: Option[Position] =
-          if self.exists then Some(self.sourcePos) else None
+          if self.exists && self.span.exists then Some(self.sourcePos) else None
```

Comment:
I investigated this and found that the only symbols without positions are the ones defined in Java and permanent symbols defined synthetically by the compiler. The current fix is fine but it could hide real bugs in the future.

An alternative is to use
```suggestion
          if self.exists && !self.lexicallyEnclosingClass.isOneOf(dotc.core.Flags.JavaDefined | dotc.core.Flags.Permanent) then
            Some(self.sourcePos)
          else
            None
```

Reply:
I propose in https://github.com/lampepfl/dotty/issues/16619 to centralise this check, to make it even less likely that we hide bugs in the future